android/src/com/frostwire/android/core/FileDescriptor.java

```diff
@@ -82,6 +84,39 @@ public FileDescriptor(int id, String artist, String title, String album, String
         ensureCorrectMimeType(this);
     }
 
+    public FileDescriptor(Bundle bundle) {
+        this(bundle.getInt("id"),
```

Comment:
I mean in reverse, having `fromBundle` not static and call it in the constructor